### PR TITLE
Adds urls.py indexer and autocomplete

### PR DESCRIPTION
**Pull request by scuml**
This one's a game-changer. 

This branch creates an index of named urls available to django and allows them to be used as autocomplete items within `{% url %}` tags and `reverse(` method calls.  This saves an incredible amount of time and mental energy flipping back and forth from `urls.py` with the name and parameters used in a url.  

Depends on comment/uncomment branch, so merging that first will clean up the 'changed files' list. 

ScreenFlow

### CODE REVIEW

**Comment by squ1b3r** [comment]
Damn, dude. You are on fire! xD

Before I do any reviews, I see there are imports that require Django. What happens if the user doesn't have it installed? Or it's installed in a virtualenv which Sublime Text has no access to?

**Comment by scuml** [comment]
The url-indexer uses the project setting: `python_interpreter` to make sure that `url_extract.py` is run through the user's virtualenv.  If django does not exist in the virtualenv, the plugin silently stops indexing.

File: commands/url_indexer.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+    def get_index_for_window(self, window):
+        return self.window_index.get(window)
+
+
+djurls_indexer = DjangoUrlsIndexer(sublime.active_window())
+
+
+class DjangoUrlsPlugin(sublime_plugin.EventListener):
+    # def on_activated(self, view):
```
Comment (squ1b3r) [inline commands/url_indexer.py:92]: Do we need this?

Comment (scuml) [reply commands/url_indexer.py]: It's helpful to quickly auto-refresh the index when debugging, but I'll remove it.

**Review by squ1b3r — APPROVED** [review]

**Comment by squ1b3r** [comment]
Thanks again for this contribution!

**Pull request merged**